pkg/providers/instance/instance.go

```diff
@@ -670,6 +705,54 @@ func (p *Provider) getAKSIdentifyingExtension() *armcompute.VirtualMachineExtens
 	return vmExtension
 }
 
+func (p *Provider) getCSExtension(cse string, isWindows bool) *armcompute.VirtualMachineExtension {
+	const (
+		vmExtensionType     = "Microsoft.Compute/virtualMachines/extensions"
+		cseNameWindows      = "windows-cse-agent-karpenter"
+		cseTypeWindows      = "CustomScriptExtension"
+		csePublisherWindows = "Microsoft.Compute"
+		cseVersionWindows   = "1.10"
+		cseNameLinux        = "cse-agent-karpenter"
+		cseTypeLinux        = "CustomScript"
+		csePublisherLinux   = "Microsoft.Azure.Extensions"
+		cseVersionLinux     = "2.0"
+	)
+
+	if isWindows { // TODO(Windows)
+		return &armcompute.VirtualMachineExtension{
+			Location: lo.ToPtr(p.location),
+			Name:     lo.ToPtr(cseNameWindows),
+			Type:     lo.ToPtr(vmExtensionType),
+			Properties: &armcompute.VirtualMachineExtensionProperties{
+				AutoUpgradeMinorVersion: lo.ToPtr(true),
+				Type:                    lo.ToPtr(cseTypeWindows),
+				Publisher:               lo.ToPtr(csePublisherWindows),
+				TypeHandlerVersion:      lo.ToPtr(cseVersionWindows),
+				Settings:                &map[string]interface{}{},
+				ProtectedSettings: &map[string]interface{}{
+					"commandToExecute": cse,
+				},
+			},
+		}
+	} else {
+		return &armcompute.VirtualMachineExtension{
+			Location: lo.ToPtr(p.location),
+			Name:     lo.ToPtr(cseNameLinux),
+			Type:     lo.ToPtr(vmExtensionType),
+			Properties: &armcompute.VirtualMachineExtensionProperties{
+				AutoUpgradeMinorVersion: lo.ToPtr(true),
+				Type:                    lo.ToPtr(cseTypeLinux),
+				Publisher:               lo.ToPtr(csePublisherLinux),
+				TypeHandlerVersion:      lo.ToPtr(cseVersionLinux),
+				Settings:                &map[string]interface{}{},
+				ProtectedSettings: &map[string]interface{}{
+					"commandToExecute": cse,
+				},
+			},
+		}
+	}
+}
```

Comment:
Something like this is more readable
```suggestion
return &armcompute.VirtualMachineExtension{
	Location: lo.ToPtr(p.location),
	Name:     lo.ToPtr(lo.Ternary(isWindows, cseNameWindows, cseNameLinux)),
	Type:     lo.ToPtr(vmExtensionType),
	Properties: &armcompute.VirtualMachineExtensionProperties{
		AutoUpgradeMinorVersion: lo.ToPtr(true),
		Type:                    lo.ToPtr(lo.Ternary(isWindows, cseTypeWindows, cseTypeLinux)),
		Publisher:               lo.ToPtr(lo.Ternary(isWindows, csePublisherWindows, csePublisherLinux)),
		TypeHandlerVersion:      lo.ToPtr(lo.Ternary(isWindows, cseVersionWindows, cseVersionLinux)),
		Settings:                &map[string]interface{}{},
		ProtectedSettings: &map[string]interface{}{
			"commandToExecute": cse,
		},
	},
}
```